### ISSUE TITLE
How can I rename an uploaded image with signed request?

### ISSUE THREAD

**Issue by Lebron1992** [cloudinary/cloudinary_ios#147]
Hi!  When I executed the following code, I got the error: `"The operation couldn’t be completed. (com.cloudinary.error error 400.)"`.

Anyone can help me? Thanks!

```
            let config = CLDConfiguration(cloudName: cloudName, apiKey: apiKey)
            let cloudinary = CLDCloudinary(configuration: config)

            let signature = CLDSignature(signature: signatureStr, timestamp: timestamp)

            let params = CLDRenameRequestParams()
            params.setSignature(signature)
            params.setResourceType(.image)

            cloudinary.createManagementApi()
                .rename(orignalPublicID,
                        to: newPublicID,
                        overwrite: true,
                        invalidate: true,
                        params: params,
                        completionHandler: { (renameResult, error) in
                            guard let result = renameResult else {
                                completion(nil, error)
                                return
                            }
                            completion(result.publicId, error)
                })
```

**Comment by aditimadan-Cloudinary** [comment]
Hi Lebron,

I will need some more information to debug this issue, can you please open a support request here: https://support.cloudinary.com/hc/en-us/requests/new with your cloud name and the image URL you tried to rename. Also, it will be great if you can share the params you passed.

Thanks,

**Comment by Lebron1992** [comment]
Sorry, for security reason, We can't share params to you. But let me describe the issue in detail: 

I can upload images with the `cloudName`, `apikey`, `signatureStr`, `timestamp` and `notification_url`, all of the params are fetched from our server. 

In the same way, I fetched all of the params from endpoint which is the same as uploading image. And then created `CLDConfiguration `, `CLDCloudinary `, `CLDRenameRequestParams `, finally I got the error after called `rename ` api.

The way to generate signature for uploading image and the way to generate signature for renaming image, are these two ways the same? Or we have to user another way to generate signature for renaming image?

Thanks!
